src/main/java/org/opensearch/neuralsearch/query/HybridQueryBuilder.java

```diff
@@ -109,8 +109,8 @@ protected void doXContent(XContentBuilder builder, Params params) throws IOExcep
             queryBuilder.toXContent(builder, params);
         }
         builder.endArray();
-        if (isClusterOnOrAfterMinReqVersionForPaginationInHybridQuery()) {
```

Comment:
do we still need method `isClusterOnOrAfterMinReqVersionForPaginationInHybridQuery` in MinClusterVersionUtil? Looks like you're using isClusterOnOrAfterMinReqVersionForPaginationInHybridQuery == false and paginationDepth == null interchangeably

Reply:
So how this works is older versions of OS when there is no pagination_depth sent then we will not set that field in the builder object. But when the same method runs for version>=2.19 then in the fromXContent method we give default value 10 when we don't get pagination_depth by the user.

Reply:
in this case it's a combination of the version check and values being not null. Should you do OR of two? What I'm essentially trying to say - we are checking one condition in one place and another condition in another scenario, and it's not that great.

Reply:
The thing is we do not use `isClusterOnOrAfterMinReqVersionForPaginationInHybridQuery` in doXContent of any other query builder. Every other place we use the same non null check. Recently expandNested param has been added by heemin in the neuralquerybuilder and it also follow the same non null check.

Reply:
Isn't there a way with StreamOutput and StreamInput to get the version that it's writing to or coming from?

Reply:
Something like this: https://github.com/opensearch-project/k-NN/blob/main/src/main/java/org/opensearch/knn/indices/ModelMetadata.java#L83
https://github.com/opensearch-project/k-NN/blob/main/src/main/java/org/opensearch/knn/indices/ModelMetadata.java#L502

Reply:
I will address this in https://github.com/opensearch-project/neural-search/issues/1097